src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -63,16 +63,45 @@ public TaskService(TaskRepository taskRepository,
         this.web3jService = web3jService;
     }
 
-    public Optional<Task> addTask(String chainDealId, int taskIndex, String imageName,
-    String commandLine, int trust, long maxExecutionTime, String tag) {
-        if (getTasksByChainDealIdAndTaskIndex(chainDealId, taskIndex).isEmpty()) {
-            log.info("Add new task [chainDealId:{}, taskIndex:{}, imageName:{}, commandLine:{}, trust:{}]",
-                    chainDealId, taskIndex, imageName, commandLine, trust);
-            return Optional.of(taskRepository.save(new Task(chainDealId, taskIndex, imageName, commandLine, trust, maxExecutionTime, tag)));
-        }
-        log.info("Task already added [chainDealId:{}, taskIndex:{}, imageName:{}, commandLine:{}, trust:{}]",
-                chainDealId, taskIndex, imageName, commandLine, trust);
-        return Optional.empty();
+    /**
+     * Save task in database if it does not
+     * already exist.
+     * 
+     * @param chainDealId
+     * @param taskIndex
+     * @param imageName
+     * @param commandLine
+     * @param trust
+     * @param maxExecutionTime
+     * @param tag
+     * @return optional containing the saved
+     * task, {@link Optional#empty()} otherwise.
+     */
+    public Optional<Task> addTask(
+            String chainDealId,
+            int taskIndex,
+            String imageName,
+            String commandLine,
+            int trust,
+            long maxExecutionTime,
+            String tag
+    ) {
+        boolean isPresent = taskRepository
+                .findByChainDealIdAndTaskIndex(chainDealId, taskIndex)
+                .isPresent();
+        if (isPresent) {
```

Comment:
Maybe we could have a full functional-like block here?

Reply:
Done, much cleaner thanks!